syncstorage/storage/sql/dbconnect.py

```diff
@@ -144,21 +159,85 @@ def _get_bso_columns(table_name):
 
 bso = Table("bso", metadata, *_get_bso_columns("bso"))
 
+# Table mapping (user_id, collection_id) => batch IDs
+
+batch_uploads = Table(
+    "batch_uploads",
+    metadata,
+    Column("batch", BigInteger, primary_key=True, nullable=False),
```

Comment:
If we're using just a timestamp as the batch id, we'll almost certainly generate spurious collisions between multiple users trying to create a transaction at the same time.  We could make this a compound primary key on `(batch, userid)`, or make `batch` a string and include more information that way.  @rtilder thoughts?